design-document.md

```diff
@@ -232,6 +232,352 @@ Note: Two non-colluding aggregators --- one leader and one standard aggregator
 --- are required to provide protect the inputs' privacy. Additional aggregators
 may be used to make the system more resilient; see {{CITE}}.
 
+## Ingestion {#proto-ingestion}
+
+[[TODO: Decide how to authenticate the leader in leader-to-helper connections.
+One option is to use client certificates for TLS; another is to have the leader
+sign its messages directly, as in Prio v2.]]
+
+Data ingestion involves the client uploading its secret shared inputs and the
+aggregators collectively validating the corresponding input. Before this process
+begins, the following must be true (the acronym "PA" stands for "Private
+Aggregation"):
+1. The client knows the URL of the leader endpoint, e.g., `example.com/metrics`.
+   We write this URL as `[leader]` below. (We write `[helper]` for the helper's
+   URL and `[aggregator]` for the URL of some aggregator.)
+1. The client has a root certificate it can use to authenticate the leader.
+1. The leader's root certificates are sufficient to authenticate each helper.
+1. The helper has chosen an HPKE key pair.
+1. The aggregators agree on the PA parameters associated with a given PA task.
+
+The PA parameters are structured as follows:
+
+```
+struct {
+  PATask task;
+  select (PAClientParam.task.proto) {
+    case prio: PrioParam;
+    case hits: HitsParam;
+  }
+} PAParam;
+
+struct {
+  PAProto proto;
+  uint16 version;
+  uint64 id;
+} PATask;
+
+Url opaque<0..255>;
+
+enum { prio(0), hits(1) } PAProto;
+```
+
+[[TODO: Add batch parameters, including min/max batch size, batch time window,
+etc.]]
+
+[[NOTE: `PAParamters.task.id` is a 64-bit integer because it may be useful for
+task IDs to be sparse.]]
+
+`PrioParam` is defined in {{prio-messages}} and `HitsParam` is defined in
+{{hits-messages}}.
+
+### Discovery
+
+To begin, the client first needs to discover the parameters and the URL of
+each helper. The client sends an HTTP request to `[leader]/parameters`. The
```

Comment:
A GET? A POST? [ACME](https://www.rfc-editor.org/rfc/rfc8555.html) does a pretty good job of describing how to talk to this kind of HTTP services server.

Reply:
I think it should be a GET here. Specified this here and elsewhere.